### PR TITLE
[pickers] Fix regression on `PickerValidDate`

### LINKED ISSUE #14895: [pickers] Cannot compile in TypeScript with `skipLibCheck: false`
### Steps to reproduce

Link to live example: (required)

https://stackblitz.com/edit/react-5frnjq?file=tsconfig.json,Demo.tsx,node_modules%2F%40mui%2Fx-date-pickers-pro%2FDateRangePicker%2Findex.d.ts,node_modules%2F%40mui%2Fx-date-pickers-pro%2FDateRangePicker%2FDateRangePickerToolbar.d.ts

Steps:
1. Open DateRangePickerToolbar.d.ts

Make sure to have `skipLibCheck` set to false in `tsconfig.json`.

### Current behavior

TypeScript compiler errors shown under `import("luxon")` and `import("moment")`

image

### Expected behavior

No TypeScript compiler errors

### Context

_No response_

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: Windows 11 10.0.22631
  Binaries:
    Node: 22.9.0 - ~\AppData\Local\fnm_multishells\2744_1728469629967\node.EXE
    npm: 10.8.3 - ~\AppData\Local\fnm_multishells\2744_1728469629967\npm.CMD
    pnpm: 9.4.0 - ~\AppData\Local\pnpm\pnpm.EXE
  Browsers:
    Chrome: Not Found
    Edge: Chromium (128.0.2739.54)
  npmPackages:
    @emotion/react: 11.13.3 => 11.13.3
    @emotion/styled: 11.13.0 => 11.13.0
    @mui/core-downloads-tracker:  6.1.2
    @mui/icons-material: 6.1.2 => 6.1.2
    @mui/material: 6.1.2 => 6.1.2
    @mui/private-theming:  6.1.2
    @mui/styled-engine:  6.1.2
    @mui/system:  6.1.2
    @mui/types:  7.2.17
    @mui/utils:  5.16.6
    @mui/x-data-grid:  7.19.0
    @mui/x-data-grid-pro: 7.19.0 => 7.19.0
    @mui/x-date-pickers:  7.19.0
    @mui/x-date-pickers-pro: 7.19.0 => 7.19.0
    @mui/x-internals:  7.18.0
    @mui/x-license:  7.18.0
    @types/react: 18.3.11 => 18.3.11
    react: 18.3.1 => 18.3.1
    react-dom: 18.3.1 => 18.3.1
    typescript: 5.6.2 => 5.6.2
```
</details>

**Search keywords**: skipLibCheck luxon moment pickers

### PR DESCRIPTION
**Pull request by flaviendelangle**
Fix #14895

### CODE REVIEW

File: packages/x-date-pickers-pro/src/DateTimeRangePicker/DateTimeRangePickerToolbar.tsx [hunk 1]
```diff
@@ -112,6 +112,10 @@ const DateTimeRangePickerToolbarEnd = styled(DateTimePickerToolbar, {
   ],
 }) as DateTimeRangePickerStartOrEndToolbarComponent;
 
+type DateTimeRangePickerToolbarComponent = (<TDate extends PickerValidDate>(
```
Comment (flaviendelangle) [inline packages/x-date-pickers-pro/src/DateTimeRangePicker/DateTimeRangePickerToolbar.tsx:115]: I don't understand why, but without it the type gets inline in the bundle and we end up with the following:

image

Comment (michelengelen) [reply packages/x-date-pickers-pro/src/DateTimeRangePicker/DateTimeRangePickerToolbar.tsx:115]: 🤷🏼

**Review by michelengelen — COMMENTED** [review]
Seems good to me

**Review by LukasTy — APPROVED** [review]
LGTM. 👍 
Other cases would result in the same output if the `as` casting is removed. 🙈 
I.e. `DatePickerToolbar`.
Ideally—it would be nice to automate the avoidance of such output. 😆

**Comment by flaviendelangle** [comment]
> Ideally—it would be nice to automate the avoidance of such output. 

A dirty solution would be to look automatically for `import('lodash')` and equivalent in the build (outside of the adapters)

**Comment by LukasTy** [comment]
> A dirty solution would be to look automatically for `import('lodash')` and equivalent in the build (outside of the adapters)

Yeah, my only idea as well. 🤔 
But, I'd say that we could leave this effort for when and if we encounter the same problem. 👍 🙈

**Pull request merged**